File: inspect_data.js
const XLSX = require('xlsx');

const filePath = "C:\\Users\\HP\\Desktop\\Sales VG30 Dashboard all india prising.xlsx";

try {
    console.log("Reading Excel headers...");
    // Read only the first 10 rows to be fast
    const workbook = XLSX.readFile(filePath, { sheetRows: 10 });
    const firstSheetName = workbook.SheetNames[0];
    const worksheet = workbook.Sheets[firstSheetName];

    // Convert to JSON to see headers
    const jsonData = XLSX.utils.sheet_to_json(worksheet, { header: 1 });

    if (jsonData.length > 0) {
        console.log("\nHeaders found:");
        console.log(jsonData[0]);
        console.log("\nSample Row 1:");
        console.log(jsonData[1]);
    } else {
        console.log("No data found in the first sheet.");
    }
} catch (error) {
    console.error("Error reading file:", error.message);
}

File: inspect_data.py
import pandas as pd
import os

file_path = r"C:\Users\HP\Desktop\Sales VG30 Dashboard all india prising.xlsx"

try:
    if not os.path.exists(file_path):
        print(f"Error: File not found at {file_path}")
        exit()

    print("Reading first 5 rows of the Excel file...")
    # Read only the first 5 rows to be fast
    df = pd.read_excel(file_path, nrows=5)
    
    print("\nColumns found:")
    for col in df.columns:
        print(f"- {col}")
        
    print("\nSample Data (First 2 rows):")
    print(df.head(2).to_string())

    print("\nData Types:")
    print(df.dtypes)

except Exception as e:
    print(f"An error occurred: {e}")
